Windows build fails: vcpkg pin references msys2 pkgconf URL that no longer exists (404)

## Summary

  On Windows, `run_cmake.bat` fails during the vcpkg `FetchContent` step because the pinned vcpkg commit references an msys2 `pkgconf` package URL that has been removed from every msys2 mirror. The build
   can't proceed past zlib's `vcpkg_fixup_pkgconfig` call.

  This affects **every active branch** in this repository that integrates vcpkg (`master`, `release-2.6.14`, `andy/planetary`, `andy/svb-sdk-1.12.1`, `bw_solarsys`). No existing issue or PR tracks this —
   searched for `vcpkg`, `pkgconf`, `msys2`, `404`.

  ## Environment

  - OS: Windows 11
  - CMake: 3.29
  - Visual Studio 2022 (17.14)
  - Branch: `master` (confirmed identical pin on other branches — see below)
  - Architecture: x86 (Win32)

  ## Reproduction

  ▎ .\run_cmake.bat

  ## Failure

  vcpkg attempts to acquire pkg-config via msys2 and hits 404 on all six mirrors:

  Downloading https://mirror.msys2.org/mingw/mingw64/mingw-w64-x86_64-pkgconf-1~2.2.0-1-any.pkg.tar.zst
  ...
  CUSTOMBUILD : error : Failed to download from mirror set
  CUSTOMBUILD : error : https://mirror.msys2.org/...status code 404
  CUSTOMBUILD : error : https://repo.msys2.org/...status code 404
  CUSTOMBUILD : error : https://mirror.yandex.ru/...status code 404
  CUSTOMBUILD : error : https://mirrors.tuna.tsinghua.edu.cn/...status code 404
  CUSTOMBUILD : error : https://mirrors.ustc.edu.cn/...status code 404
  CUSTOMBUILD : error : https://mirror.selfnet.de/...status code 404

  CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
    Failed to download file with error: 1
  ...
  Call Stack:
    scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake:52 (vcpkg_acquire_msys)
    ...
    scripts/cmake/vcpkg_fixup_pkgconfig.cmake:193 (vcpkg_find_acquire_program)
    ports/zlib/portfile.cmake:43 (vcpkg_fixup_pkgconfig)

  CUSTOMBUILD : error : building zlib:x86-windows failed with: BUILD_FAILED

  ## Root cause

  `thirdparty/thirdparty.cmake` pins vcpkg at commit `f7423ee180c4b7f40d43402c2feb3859161ef625` (2024-06-10). That snapshot of `scripts/cmake/vcpkg_find_acquire_program(PKGCONFIG).cmake` references
  `mingw-w64-x86_64-pkgconf-1~2.2.0-1`, which msys2 has since rotated out of all mirrors.

  Confirmed live vs. dead versions on `mirror.msys2.org`:

  | pkgconf version | Status |
  |---|---|
  | `1~1.8.0-2` (used on `andy/svb-sdk-1.12.1`) | 404 |
  | `1~2.2.0-1` (used on `master` and other branches) | 404 |
  | `1~2.3.0-1` | 200 |
  | `1~2.4.3-1` | 200 |
  | `1~2.5.1-1` | 200 |

  vcpkg fixed this in microsoft/vcpkg#41665 (commit `359a5bb0c`, 2024-10-25), which first appeared in release tag `2024.11.16`.

  ## Affected branches

  | Branch | vcpkg pin | Status |
  |---|---|---|
  | `master` | `f7423ee` (2024-06-10) | Broken — pkgconf 2.2.0-1 (404) |
  | `release-2.6.14` | `f7423ee` | Broken — same pin |
  | `andy/planetary` | `f7423ee` | Broken — same pin |
  | `bw_solarsys` | `f7423ee` | Broken — same pin |
  | `andy/svb-sdk-1.12.1` | `61f610845` (2023-11-11) | Broken — older pin, pkgconf 1.8.0-2 also 404 |
  | `release/v2.6.13` | (no vcpkg integration) | n/a |

  The `andy/svb-sdk-1.12.1` branch has been broken longer, since its even older vcpkg pin references a pkgconf version that was rotated out earlier.

  ## Suggested fix

  Bump the `GIT_TAG` in `thirdparty/thirdparty.cmake` from the current pinned commit to at least vcpkg `2024.11.16`:

  ```cmake
  FetchContent_Declare(
    vcpkg
    GIT_REPOSITORY https://github.com/microsoft/vcpkg.git
    GIT_TAG 2024.11.16   # was: f7423ee180c4b7f40d43402c2feb3859161ef625
    ...
  )

  This is the minimum bump that restores the build. Direct dependency drift from the bump:

  - cfitsio 3.49#5 → 3.49#6 (port revision only — same upstream source)
  - curl 8.8.0 → 8.11.0 (patch-level)
  - eigen3 3.4.0 (unchanged)
  - opencv4 4.8.0 → 4.10.0 (minor — ABI-stable within 4.x)
  - zlib 1.3.1 (unchanged)

  DLL filenames hardcoded in thirdparty.cmake are major-version-tied and remain identical (opencv_*4.dll, jpeg62.dll, libpng16.dll, zlib1.dll, libcurl.dll, cfitsio.dll), so no other CMake edits are
  required.

  For andy/svb-sdk-1.12.1, the same bump applies — that branch's older pin needs to move forward at least as far.

  Note

  The same issue will likely recur whenever msys2 rotates the next pkgconf version out, unless vcpkg pins are kept reasonably current. Considering a maintenance policy of bumping the vcpkg pin once or
  twice a year would prevent re-occurrence.

--- a/thirdparty/thirdparty.cmake
+++ b/thirdparty/thirdparty.cmake
@@ -73,7 +73,7 @@ if(WIN32)
   FetchContent_Declare(
     vcpkg
     GIT_REPOSITORY https://github.com/microsoft/vcpkg.git
-    GIT_TAG f7423ee180c4b7f40d43402c2feb3859161ef625
+    GIT_TAG 84bab45d415d22042bd0b9081aea57f362da3f35  # 2025.12.12
     UPDATE_COMMAND bootstrap-vcpkg.bat -disableMetrics
     COMMAND ${CMAKE_COMMAND} -E echo "Building vcpkg cfitsio"
     COMMAND vcpkg install --binarysource=default --no-print-usage cfitsio:${WINDOWS_ARCH}-windows
@@ -92,7 +92,7 @@ if(WIN32)
   set(VCPKG_DEBUG_LIB ${VCPKG_PREFIX}/debug/lib)
   set(VCPKG_RELEASE_LIB ${VCPKG_PREFIX}/lib)
   set(VCPKG_INCLUDE ${VCPKG_PREFIX}/include)
-  include_directories(${VCPKG_INCLUDE})
+  include_directories(${VCPKG_INCLUDE} ${VCPKG_INCLUDE}/opencv4)
 endif()
 
 if(APPLE)

